File: backend/src/main/java/com/teamportal/service/TeamService.java
package com.teamportal.service;

import com.teamportal.dto.TeamStatsDTO;
import com.teamportal.repository.TeamRepository;
import org.springframework.stereotype.Service;

@Service
public class TeamService {
    private final TeamRepository teamRepository;

    public TeamService(TeamRepository teamRepository) {
        this.teamRepository = teamRepository;
    }

    public TeamStatsDTO getTeamStats(String teamName) {
        var team = teamRepository.findByNameIgnoreCase(teamName)
                .orElseThrow(() -> new RuntimeException("Team not found: " + teamName));

        return new TeamStatsDTO(
                team.getMembers(),
                team.getActiveProjects(),
                team.getCompletedThisMonth(),
                team.getEfficiency()
        );
    }
}

File: backend/src/main/kotlin/com/teamportal/service/TeamService.kt
package com.teamportal.service

import com.teamportal.repository.TeamRepository
import org.springframework.stereotype.Service

data class TeamStatsDTO(
    val members: Int,
    val activeProjects: Int,
    val completedThisMonth: Int,
    val efficiency: Int
)

@Service
class TeamService(
    private val teamRepository: TeamRepository
) {
    fun getTeamStats(teamName: String): TeamStatsDTO {
        val team = teamRepository.findByNameIgnoreCase(teamName) ?: throw RuntimeException("Team not found: $teamName")

        return TeamStatsDTO(
            members = team.members,
            activeProjects = team.activeProjects,
            completedThisMonth = team.completedThisMonth,
            efficiency = team.efficiency
        )
    }
}
